solution.rb

```diff
@@ -152,51 +125,26 @@
             ]
         }
       ]
-      
-      
-      coffee_average_per_person = []
-    
-    # Q_3
-    # ```
-    #   #######The solution by using .each loop#######
-    # people.each do |hash|  ##first loop for hash 
-    #     new_person = {name:"",coffee_average:0}
-    #     new_person[:name] = hash[:name]
-    #     puts hash[:name]
-    #     counter = 0
-    #     total_amout = 0
-    #     hash[:transactions].each do |key,value|  ##second loop for transcations [{},{},{}]
-    #         if key[:type] == "COFFEE"
-    #             total_amout += key[:amount]
-    #             counter += 1
-    #         end
+      avrage = 0
+      sum = 0
+    people.each do |name|
+      names =  name[:name]
+    end
+    people.each do |trans|
+    transa = trans[:transactions]  
+     transa.each do |avg|
+         if avg[:type] == "COFFEE"
+          sum =  avg[:amount] + avrage
```

Comment:
good job; but as we know the average is the sum of the numbers divided by the total number of values in the set; so we need two value one for the ```sum``, and the other for the ```total_number_of_values``` in the set;
```rb
people.each_with_index do |trans, i|
  total_number_of_values = 0 
  sum = 0
  transa = trans[:transactions]  
  transa.each do |avg|
    if avg[:type] == "COFFEE"
    total_number_of_values += 1
      sum +=  avg[:amount] 
    end   
  end
  average = sum / total_number_of_values
  puts "average coffee of " + names[i] + " is " + average.to_s
end

```